spec/hatchet/profile_d_scripts_spec.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require_relative '../spec_helper'
+
+RSpec.describe '.profile.d/ scripts' do
+  it 'sets the required run-time env vars' do
+    Hatchet::Runner.new('spec/fixtures/procfile', run_multi: true).deploy do |app|
+      # These are written as a single test to reduce end to end test time. This repo uses parallel_split_test,
+      # so we can't perform app setup in a `before(:all)` and have multiple tests run against the single app.
+
+      list_envs_cmd = 'env | sort | grep -vE "^(_|DYNO|PORT|PS1|SHLVL|TERM)="'
```

Comment:
I would whitelist the env vars, just in case it ever fails and for whatever reason exposes future variables that you don't want to be in CI run logs.

Reply:
I agree there are trade-offs of both approaches. 

The reason I had chosen an exclusion list, was that I want to make sure any env vars added to the buildpack implementation in the future are caught by the test - and in the case of intentionally added env vars this will ensure test coverage is added, or in the case of accidentally exported internal vars, this will prevent shipping them by mistake.

The downsides to an exclusion list approach are the chance of (a) test failures/churn if the platform starts setting additional env vars, or (b) the chance of any new secret env vars. However, for (a) in general the approach I've taken for the test fixtures is to favour test coverage over churn risk (and in this case the frequency is going to be very rare), and for (b), it would be quite a significant platform change if all apps suddenly gained sensitive secrets in their env without warning or needing to add a addon/enable a labs etc (which would require much notification to customers, let alone deliberation internally), plus the Hatchet test accounts are standard non-privileged test users that have no additional access.

I did try out a loose regex based inclusion list approach:

```
env | sort | grep -E "^(DYNO_RAM|FORWARDED_ALLOW_IPS|GUNICORN.*|LANG|.*PATH.*|PIP.*|PYTHON.*|WEB_CONCURRENCY)="
```

However, as can be see there isn't really a consistent pattern of naming of env vars, so we can never be sure that the test isn't missing a new env var.